source/_posts/2015-11-16-immutable-strings-in-ruby-2-dot-3.markdown

```diff
@@ -0,0 +1,136 @@
+---
+layout: post
+title: "Immutable strings in Ruby 2.3"
+category:
+date: 2015-11-16 08:39:25 -0300
+comments: true
+author:
+  name: Alexis Mas
+  email: alexis@wyeworks.com
+  twitter_handle: _axxiss
+  github_handle: axxiss
+  image: /images/team/
+  description: Software Engineer. Turning coffee into software since 2008.
+---
+
+String literals immutability is one of the new features you can find in [Ruby 2.3][1].
+But what does this implies?
+
+* Equality comparison affected
+* Better performance
+* Thread safety
+
+
+Currently strings immutability is included as opt-in, as the first step of a multi
+stage rollout and if it works well it will be [enabled by default on Ruby 3.0][2].
+So it still can end up being just an experiment. Time will tell.
+
+<!--more-->
+
+# Immutability
+
+Quoting [wikipedia][3]:
+
+> An immutable object is an object whose state cannot be modified after it is created.
+
+As immutable objects state cannot be changed it doesn't make sense to copy the
+same object over and over. Following that approach it will have three major
+consequences:
```

Comment:
I'm not sure if I understood well the idea of this section.

@Axxiss are you trying to say something like the following? (just to double check the message meaning)

> As immutable objects state cannot be changed it doesn't make sense to copy the same object over and over. In fact, Ruby 2.3 will hold a unique instance for each string literal value used in the program. The programmer is able to have as many references to this instances as needed.

> This approach will help to avoid unnecessary data duplication, therefore unnecessary object allocation is prevented and it will imply that the program runs faster because it will spend less time in the Garbage Collector.

> Also, _insert here short explanation about the benefits of inmutable strings for multi-thread programs_

One thing that I discovered rewording the paragraph is the fact that 'Data duplication' and 'Unnecessary object allocation' are more or less the same issue (cause and consequence relationship) so I would not present them as two different items in a bullet list.